tools/config/__init__.py

```diff
@@ -393,7 +393,7 @@ def format_validation_error(self, error, path):
             return self.format_validation_error(error.context[0], path)
         else:
             return "in {} element {}: {}".format(
-                path, str(".".join(str(p) for p in error.absolute_path)), error.message)
+                path, str(".".join(str(p) for p in error.absolute_path)), error.message).replace('u\'', '\'')
```

Comment:
I'd like to see this done another way, possibly without the outer `str`.

Reply:
```diff
diff --git a/tools/config/__init__.py b/tools/config/__init__.py
index 04677215e..cbfc980af 100644
--- a/tools/config/__init__.py
+++ b/tools/config/__init__.py
@@ -393,7 +393,7 @@ class Config(object):
             return self.format_validation_error(error.context[0], path)
         else:
             return "in {} element {}: {}".format(
-                path, str(".".join(str(p) for p in error.absolute_path)), error.message)
+                path, ".".join(str(p) for p in error.absolute_path), error.message)
 
     def __init__(self, tgt, top_level_dirs=None, app_config=None):
         """Construct a mbed configuration
```

Reply:
I feel like I'm missing something. Implementing the above diff does not appear to work across Py versions.

In Py2, the error returned looks like:
`[<Validation Error: "Additional properties are not allowed (u'unknown_key' was unexpected)">]`

In Py3:
`[<Validation Error: "Additional properties are not allowed ('unknown_key' was unexpected)">]`

Reply:
Oh dang it. That's a string that's coming directly from jsonschema. We may have to allow both versions...

Reply:
Also, for reference, `error.message` is typed as a string in this function, which from what I understand, the `u'` is already a part of the variable.

Reply:
>Oh dang it. That's a string that's coming directly from jsonschema. 

Yeah. It does it's own internal conversion to support both Py versions.

>We may have to allow both versions...

Are you referring to having to support both within json parsing, or just for the particular tests?

Reply:
Alright, can we move the `.replace` to the `error.message` then? I don't want to accidentally replace more than we want to.

Reply:
EH, not that important.